src/metrics/sol/startSubscription.ts

```diff
@@ -35,12 +35,19 @@ export const startSubscription = async (context: Context) => {
     try {
         subscriptionId = connection.onLogs(
             new PublicKey(global.solanaCurrentOnChainKey),
-            (log: any) => {
+            async (log: any) => {
                 if (log.err !== null) {
-                    metricSolanaTxReverted.labels(log.signature).set(1);
-                    setTimeout(() => {
-                        metricSolanaTxReverted.remove(log.signature);
-                    }, 600000); // 10m
+                    const transaction = await context.connection.getTransaction(log.signature, {
+                        commitment: 'finalized',
```

Comment:
the subscription doesn't seem to be on finalized, right? If so then it probably make sense to not set the commitment to finalized here either.

Reply:
Yes it is on finalized!